services/frontend-service/src/ui/components/ProductVersion/ProductVersion.tsx

```diff
@@ -15,40 +15,108 @@ along with kuberpult. If not, see <https://directory.fsf.org/wiki/License:Expat>
 Copyright 2023 freiheit.com*/
 import './ProductVersion.scss';
 import * as React from 'react';
-import { refreshTags, useTags } from '../../utils/store';
+import { refreshTags, useTags, getSummary, useSummaryDisplay } from '../../utils/store';
+import { DisplayManifestLink, DisplaySourceLink } from '../../utils/Links';
+import { Spinner } from '../Spinner/Spinner';
+import { ProductSummary } from '../../../api/api';
 
 export type ProductVersionProps = {
     environment: string;
 };
+
 export const ProductVersion: React.FC<ProductVersionProps> = (props) => {
     React.useEffect(() => {
+        setShowSpinner(true);
         refreshTags();
+        setShowSpinner(false);
     }, []);
     const { environment } = props;
     const [open, setOpen] = React.useState(false);
-    const openClose = React.useCallback(() => {
-        setOpen(!open);
-    }, [open, setOpen]);
+    const openClose = React.useCallback(
+        (e: React.ChangeEvent<HTMLSelectElement>) => {
+            setShowSpinner(true);
```

Comment:
I cannot test it right now, but I doubt that this has the intended effect, because `getSummary` does not wait for the response (e.g. it's not returning a Promise). 
Did you try to enable network throttling in chrome to test this (manually)?

For comparison, we are doing something similar in the function `useFrontendConfig` with the `configReady` flag.

Reply:
I tested it locally, the spinner does not show up for me.

Reply:
(to be precise, the spinner does show up, but it's not the right one. It never says "loading tag data")

Reply:
Fixed and tested with network throttling